src/helpers/videoUrlHelper.test.js

```diff
@@ -0,0 +1,293 @@
+/* eslint-disable no-unused-vars */
+import { videoUrlHelper, checkIfValidVideoLink } from './videoUrlHelper';
+import config from '@plone/volto/registry';
+
+const { settings } = config;
+
+const YT_TEST_ID = 'dQw4w9WgXcQ';
+const VIMEO_TEST_ID = '346762373';
+
+describe('videoUrlHelper', () => {
+  it('recognizes valid video links', () => {
+    const ytValid = checkIfValidVideoLink(
+      'https://www.youtube.com/watch?v=dQw4w9WgXcQ',
+    );
+    const vimeoValid = checkIfValidVideoLink(
+      'https://www.player.vimeo.com/albums/346762373/comments',
+    );
+    const internalValid = checkIfValidVideoLink('http://localhost:3000/video');
+    expect(ytValid).toBe(true);
+    expect(vimeoValid).toBe(true);
+    expect(internalValid).toBe(true);
+  });
+  it('recognizes invalid video links', () => {
+    const ytValid = checkIfValidVideoLink(
+      'https://www.youtube.com/watch?dQw4w9WgXcQ',
+    );
+    const vimeoValid = checkIfValidVideoLink(
+      'https://www.player.vimeo.com/vid/346762373/comments',
+    );
+    const internalValid = checkIfValidVideoLink('https://www.google.it/video');
+    expect(ytValid).toBe(false);
+    expect(vimeoValid).toBe(false);
+    expect(internalValid).toBe(false);
+  });
+  it('allows whatever external links if allowExternal setting is enabled', () => {
+    const internalValid = checkIfValidVideoLink(
+      'https://www.google.it/video',
+      true,
+    );
+    expect(internalValid).toBe(true);
+  });
+  it('disallows external links if allowExternal setting is not enabled', () => {
+    const internalValid = checkIfValidVideoLink(
+      'https://www.google.it/video',
+      false,
+    );
+    expect(internalValid).toBe(false);
+  });
+  it('resolves basic watch yt link', () => {
+    const [videoId, _] = videoUrlHelper(
+      'https://www.youtube.com/watch?v=dQw4w9WgXcQ',
```

Comment:
Per rendere più chiaro il test e semplificare eventuali rifattorizzazioni future del test, qui (e anche in quelli successivi) lo farei con una stringa concatenata con la variabile YT_TEST_ID